src/Plugin/Field/FieldFormatter/StrawberryBaseFormatter.php

```diff
@@ -0,0 +1,246 @@
+      'iiif_base_url_internal' => \Drupal::config('format_strawberryfield.iiif_settings')->getOriginal('int_server_url', FALSE),
+      ] + parent::defaultSettings();
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function IiifSettingsForm(array $form, FormStateInterface $form_state) {
+    return [
+      'iiif_base_url' => [
+        '#type' => 'url',
+        '#title' => $this->t('Base Public accesible URL of your IIIF Media Server'),
+        '#default_value' => $this->getSetting('iiif_base_url'),
+        '#required' => TRUE
+      ],
+      'iiif_base_url_internal' => [
+        '#type' => 'url',
+        '#title' => $this->t('Base URL of your IIIF Media Server accesible from inside this Webserver'),
+        '#default_value' => $this->getSetting('iiif_base_url_internal'),
+        '#required' => TRUE,
+        '#element_validate' => [[$this, 'validateIiifUrl']],
+      ],
+      'iiif_reset_button' => [
+        '#type' => 'button',
+        '#name' => 'iiif_reset_button',
+        '#value' => t('Reset to IIIF Defaults'),
+        '#executes_submit_callback' => TRUE,
+        '#submit' => [[$this, 'resetIiifDefaults']],
+      ],
+    ];
+  }
+
+
+  public function validateIiifUrl(array $form, FormStateInterface $form_state) {
+    // todo: @marlo -- also broken, same issue with non-updated form_state
+    error_log(var_export( $form_state->get('iiif_base_url_internal'), true));
+    if ($form_state->get('iiif_base_url_internal')) {
+    try {
+      $response = $this->httpClient
+        ->head(
+          $form_state->get('iiif_base_url_internal')
+        );
+    }
+    catch(ConnectException $exception) {
+      $responseMessage = $exception->getMessage();
+      $form_state->setErrorByName('iiif_base_url_internal', t("We could not contact your Public IIIF server: @error", [
+        '@error' => $responseMessage
+      ]));
+    }
+    catch(ClientException $exception) {
+      $responseMessage = $exception->getMessage();
+      $form_state->setErrorByName('iiif_base_url_internal', t("We could not contact your Public IIIF server: @error", [
+        '@error' => $responseMessage
+      ]));
+    }
+    catch (ServerException $exception) {
+      $responseMessage = $exception->getMessage();
+      $form_state->setErrorByName('iiif_base_url_internal', t("We could not contact your Public IIIF server: @error", [
+        '@error' => $responseMessage
+      ]));
+    }
+    }
+  }
+
+
+  /**
+   * {@inheritdoc}
+   */
+  public function resetIiifDefaults(array $form, FormStateInterface $form_state) {
+    $this->setSettings([
+      'iiif_base_url' => $this->iiif_pub_server,
+      'iiif_base_url_internal' => $this->iiif_int_server
+    ]);
+    //todo: @marlo - this is broken :(
+    $form_state->set('iiif_base_url', $this->iiif_pub_server);
+    $form_state->set('iiif_base_url_internal', $this->iiif_int_server);
+    $form_state->setRebuild(true);
+    return $form;
```

Comment:
Nope. You should not be returning anothing on a Form submit handler (that is Drupal 7). And the signature is wrong
This is the right one `(array &$form, FormStateInterface $form_state)`